client/src/pages/SignUp.tsx

```diff
@@ -1,60 +1,48 @@
 import { useEffect, useState } from 'react';
 import { useAuthState } from 'react-firebase-hooks/auth';
-import { auth, collections } from '../firebase/connection';
+import { auth } from '../firebase/connection';
 import { useNavigate } from 'react-router-dom';
 import Form, { FormOptions } from '../components/Form';
 import Header from '../components/Header';
-import { deleteDoc, doc, setDoc } from '@firebase/firestore';
 import React from 'react';
-import { createUserWithEmailAndPassword, deleteUser, updateProfile } from '@firebase/auth';
 import ErrorModal, { ErrorInfo } from '../components/ErrorModal';
 import { signUpErrors } from '../consts/errorMessages';
 import BackButton from '../components/BackButton';
+import { getFunctions, httpsCallable } from 'firebase/functions';
+import { signInWithEmailAndPassword } from 'firebase/auth';
 
 const SignUpPage = () => {
   const navigate = useNavigate();
   const [user] = useAuthState(auth);
-  const [stage, setStage] = useState<'start' | 'updating' | 'end'>('start');
   const [error, setError] = useState<ErrorInfo>();
 
   const handleFormSubmit = async ({ name, email, password }: FormOptions) => {
-    setStage('updating');
     try {
-      const user = await createUserWithEmailAndPassword(auth, email!, password!);
-
-      if (user) {
-        try {
-          await setDoc(doc(collections.users, user.user.uid), { name: name!, acceptedTerms: true });
-        } catch (e: unknown) {
-          await deleteUser(user.user);
-          throw { code: 'error' };
-        }
-        try {
-          await updateProfile(user.user, { displayName: name });
-        } catch (e: unknown) {
-          await deleteDoc(doc(collections.users, user.user.uid));
-          await deleteUser(user.user);
-          throw { code: 'error' };
-        }
-        setStage('end');
-      }
+      const functions = getFunctions();
+      const signUp = httpsCallable(functions, 'signUp');
+      await signUp({ password, email, name });
+      await signInWithEmailAndPassword(auth, email!, password!);
     } catch (e: unknown) {
-      const error = e as { code: string };
-      if (error.code === 'auth/email-already-in-use') {
+      const error = e as { message: string; code: string };  
+      
+      if (error.code === 'functions/already-exists') {
         setError(signUpErrors.userAlreadyExists);
+      } else if (error.code === 'functions/invalid-argument') {
+        setError({ title: 'קלט לא תקין', content: error.message });
```

Comment:
Better to add new signUpErrors constants as previously done

Reply:
These are not constants as the content is based on the error.message